### PR TITLE
oreo-cursors-plus: init at unstable-2023-06-05

### PR DESCRIPTION
**Pull request by MichaelBrunn3r**
## Description of changes
Adding new cursor theme package `oreo-cursors-plus` from [https://github.com/Souravgoswami/oreo-cursors](https://github.com/Souravgoswami/oreo-cursors).
This is a fork of an older project but with more features (original: [https://github.com/varlesh/oreo-cursors](https://github.com/varlesh/oreo-cursors)).
Both repos have the same name `oreo-cursors`, thus I opted for the suffix `-plus` for the more recent one. I'm in contact with the author to make sure the name is acceptable.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/oreo-cursors-plus/default.nix
Comment (FliegendeWurst) [inline pkgs/data/icons/oreo-cursors-plus/default.nix:7]:
```suggestion
```

File: pkgs/data/icons/oreo-cursors-plus/default.nix
Comment (FliegendeWurst) [inline pkgs/data/icons/oreo-cursors-plus/default.nix:62]: The license should be either `gpl2Plus` or `gpl2Only`.

**Comment by FliegendeWurst** [comment]
Currently the package definition is not referenced anywhere. I recommend moving it to the `pkgs/by-name` directory structure. (Alternatively, you would need to add it to `pkgs/top-level/all-packages.nix`.)

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:11]:
```suggestion
    version = "unstable-2023-06-05";
```
This is the usual naming scheme for untagged versions.

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:21]:
```suggestion
    nativeBuildInputs = [ ruby inkscape xorg.xcursorgen ];
```

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:5]:
```suggestion
  ruby,
  inkscape,
  xorg,
```

**Review by FliegendeWurst — COMMENTED** [review]
LGTM overall :+1:

Result of `nixpkgs-review pr 258995` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>oreo-cursors-plus</li>
  </ul>
</details>

**Comment by D3vil0p3r** [comment]
@MichaelBrunn3r any news about this PR?

**Comment by MichaelBrunn3r** [comment]
Oh, did I forget anything? I just thought FliegendeWurst helped me to make it acceptable and now its just waiting for the next NixOS release/backport.

**Comment by D3vil0p3r** [comment]
> @D3vil0p3r Oh, did I forget anything? I just thought FliegendeWurst helped me to make it acceptable and now its just waiting for the next NixOS release/backport.

@MichaelBrunn3r there are still the suggestions of @FliegendeWurst to commit. After them, you must change the name of the Pull Request properly by changing the version with the suggested one, and finally you must squash all the commits in order to keep only the first one with the message "oreo-cursors-plus: init at unstable-2023-06-05".

**Comment by MichaelBrunn3r** [comment]
@D3vil0p3r This should be it. Sry for the trouble, I don't do this often (as can be seen by the public force push :) )
I squashed all commits regarding the package into one. However, I kept the maintainer commit, as that should be required by CI

**Review by D3vil0p3r — APPROVED** [review]
LGTM

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:10]: Filewide `with lib;` statement can be somewhat regarded as an antipattern. I do not think `lib` is used enough here to warrant it. No hard requirement though.

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:11]: I think the `echo "${cursorsConf}"` seems a bit easy to break. What happens if there are newlines? How about we create a new file instead, and overwrite the previous file if there's any content?

```suggestion
let
  newCursorsConf = writeText "oreo-cursors-plus.conf" cursorsConf;
in
  stdenv.mkDerivation rec {
```

Then we can use `cp ${newCursorsConf} ./cursors.conf` later

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:34]: See earlier suggestion about `newCursorsConf`

```suggestion
    configurePhase = ''
      runHook preConfigure
      
      ${lib.optionalString (cursorsConf != null) ''
        cp ${newCursorsConf} cursors.conf
      ''}
      
			runHook postConfigure
    '';
```

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:53]: While I appreciate the attempt at making the package more easy to debug, I think the logging is a bit over the top. The user will probably realize whether their custom configuration worked or not based on whether it needs to recompile or not, let alone the resulting derivation. Unsure about the export, but it seemed reasonable?

```suggestion
    buildPhase = ''
      runHook preBuild
      
      ${lib.optionalString (cursorsConf != null) ''
        rm -r {dist,src/oreo_*}
        export HOME=$TMP
        ruby generator/convert.rb
        make build
      ''}
      
      runHook postBuild
    '';
```

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:58]: Maybe something like this would be possible?

```suggestion
    installPhase = ''
      runHook preInstall
    
      mkdir -p $out/share
      mv ./dist $out/share/icons
      
      runHook postInstall
    '';
```

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:23]: I'm not entirely confident this works, but it would be cool to reduce the closure size if no custom build is needed.

```suggestion
    nativeBuildInputs = lib.optionals (cursorsConf != null) [ ruby inkscape xorg.xcursorgen ];
```

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:11]: This `rec` doesn't seem to be used

```suggestion
  stdenv.mkDerivation {
```

Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:11]: nit: the whole `mkDerivation` call can probably be dedented.

File: pkgs/by-name/or/oreo-cursors-plus/package.nix
Comment (h7x4) [inline pkgs/by-name/or/oreo-cursors-plus/package.nix:3]: Maybe you could try replacing `stdenv` with `stdenvNoCC`? I don't think this should require a C toolchain.

**Review by h7x4 — COMMENTED** [review]
Hello, welcome to nixpkgs! Here's a few suggestions

**Comment by MichaelBrunn3r** [comment]
I ran `nix-build --expr '(import ./. {}).oreo-cursors-plus.override {cursorsConf = "test = #123456"}'` to test building with an overriden `cursorsConf`.
Still works 👍

**Review by h7x4 — APPROVED** [review]

**Pull request merged**
